lib/parseGLTF.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var fs = require('fs');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = parseGLTF;
+
+function parseGLTF(dir, gltfData, callback) {
+    var gltf = JSON.parse(gltfData);
+    gltf.loadedShaders = {};
+
+    var shaders = gltf.shaders;
+    if (defined(shaders)) {
+        for (var shaderId in shaders) {
+            if (shaders.hasOwnProperty(shaderId)) {
+                var uri = shaders[shaderId].uri;
+                loadURI(gltf, dir, uri, 'loadedShaders', callback);
+            }
+        }
+    }
+
+    return gltf;
+}
+
+function loadURI(gltf, dir, uri, loadId, callback) {
+    var uriPath = dir + '\\' + uri;
+    fs.readFile(uriPath, function (err, data) {
+        if (err) {
+            throw err;
+        }
+
+        gltf[loadId][uri] = data;
```

Comment:
I thought more about this: even though we don't have to, let's put this in an `extras` object, e.g.,
```
shader.extras.source
```
See https://github.com/KhronosGroup/glTF/blob/master/specification/README.md#reference-shader